client/ayon_core/host/interfaces/workfiles.py

```diff
@@ -0,0 +1,1177 @@
+
+
+class IWorkfileHost:
+    """Implementation requirements to be able to use workfiles utils and tool.
+
+    Some of the methods are pre-implemented as they generally do the same in
+        all host integrations.
+
+    """
+    @abstractmethod
+    def save_workfile(self, dst_path: Optional[str] = None):
+        """Save the currently opened scene.
+
+        Args:
+            dst_path (str): Where the current scene should be saved. Or use
+                the current path if 'None' is passed.
+
+        """
+        pass
+
+    @abstractmethod
+    def open_workfile(self, filepath: str):
+        """Open passed filepath in the host.
+
+        Args:
+            filepath (str): Path to workfile.
+
+        """
+        pass
+
+    @abstractmethod
+    def get_current_workfile(self) -> Optional[str]:
+        """Retrieve a path to current opened file.
+
+        Returns:
+            Optional[str]: Path to the file which is currently opened. None if
+                nothing is opened.
+
+        """
+        return None
+
+    def workfile_has_unsaved_changes(self) -> Optional[bool]:
+        """Currently opened scene is saved.
+
+        Not all hosts can know if the current scene is saved because the API
+            of DCC does not support it.
+
+        Returns:
+            Optional[bool]: True if scene is saved and False if has unsaved
+                modifications. None if can't tell if workfiles has
+                modifications.
+
+        """
+        return None
+
+    def get_workfile_extensions(self) -> list[str]:
+        """Extensions that can be used to save the workfile to.
+
+        Notes:
+            Method may not be used if 'list_workfiles' and
+                'list_published_workfiles' are re-implemented with different
+                logic.
+
+        Returns:
+            list[str]: List of extensions that can be used for saving.
+
+        """
+        return []
+
+    def save_workfile_with_context(
+        self,
+        filepath: str,
+        folder_entity: dict[str, Any],
+        task_entity: dict[str, Any],
+        *,
+        version: Optional[int],
+        comment: Optional[str] = None,
+        description: Optional[str] = None,
+        rootless_path: Optional[str] = None,
+        workfile_entities: Optional[list[dict[str, Any]]] = None,
+        project_settings: Optional[dict[str, Any]] = None,
+        project_entity: Optional[dict[str, Any]] = None,
+        anatomy: Optional["Anatomy"] = None,
+    ):
+        """Save the current workfile with context.
+
+        Notes:
+            Should this method care about context change?
+
+        Args:
+            filepath (str): Where the current scene should be saved.
+            folder_entity (dict[str, Any]): Folder entity.
+            task_entity (dict[str, Any]): Task entity.
+            version (Optional[int]): Version of the workfile.
+            comment (Optional[str]): Comment for the workfile.
```

Comment:
```suggestion
            comment (Optional[str]): Comment for the workfile.
                Usually used in the filename template.
```
Or is this data also in the workfile info on server?